packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -501,7 +494,16 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
             }
         }
 
-        const id = await this.odspSummaryUploadManager.writeSummaryTree(summary, context);
+        const uploadManager = await this.odspSummaryUploadManagerPromise.then((m) => new m(
```

Comment:
From efficiency POV you, it makes sense to
- not load chunk in constructor, but do it here, lazily (i.e. on first summary).
- start the process before we call flushCallback(), but await result after it's done (i.e. here).
That will result in smallest impact on other parts (including boot of client, and in general - clients that will not summarize), but at the same time will have minimal impact on summary speed.

Reply:
+1 to not doing it on the constructor, we definitely want to avoid downloading + parsing + compiling this code on the hot path. @aboktor  has been looking at reducing these kinds of patterns of eagerly downloading chunks that aren't needed yet